my-app/src/pages/Form.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import axios from "axios";
+import toast, { Toaster } from "react-hot-toast";
+import "./Form.css";
+
+function Form() {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm();
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const onSubmit = async (data) => {
+    setIsSubmitting(true);
+
+    try {
+      const response = await axios.post(
```

Comment:
Better to make a hook here and pass the values as reference. Dont need to update now but for future reference.
Look into useQuery by Tanstack aswell